common/views/components/MediaObjectBase/MediaObjectBase.tsx

```diff
@@ -0,0 +1,152 @@
+import { trackEvent } from '../../../utils/ga';
+import DateRange from '../DateRange/DateRange';
+import EventDateRange from '../EventDateRange/EventDateRange';
+import StatusIndicator from '../StatusIndicator/StatusIndicator';
+import LabelsList from '../LabelsList/LabelsList';
+import ImagePlaceholder from '../ImagePlaceholder/ImagePlaceholder';
+// $FlowFixMe(tsx)
+import PartNumberIndicator from '../PartNumberIndicator/PartNumberIndicator';
+import ImageType from '../Image/Image';
+import { ColorSelection } from '../../../model/color-selections';
+import Space from '../styled/Space';
+import styled from 'styled-components';
+
+type Props = {
+  url: string | null,
+  title: string,
+  labels: ComponentProps<typeof LabelsList>,
+  description: string | ReactElement | null,
+  urlOverride: string | null,
+  extraClasses?: string,
+  partNumber: number | null,
+  color: ColorSelection | null,
+  Image: ReactElement<typeof ImageType | typeof ImagePlaceholder> | null,
+  DateInfo: ReactElement<typeof DateRange> | ReactElement<typeof EventDateRange> | null,
+  StatusIndicator: ReactElement<typeof StatusIndicator> | null,
+  ExtraInfo?: ReactNode | null,
+  xOfY: { x: number, y: number },
+  OverrideImageWrapper?: ReactNode,
+  OverrideTextWrapper?: ReactNode,
+  OverrideTitleWrapper?: ReactNode,
+  onClick?: () => void,
+};
+
+const BaseImageWrapper = styled.div.attrs(props => ({
+  className: grid({ s: 3, m: 3, l: 3, xl: 3 }),
+}))``;
+
+const BaseTitleWrapper = styled.div.attrs(props => ({
+  className: classNames({
+    'card-link__title': true,
+    [font('wb', 5)]: true,
+  })
+}))``;
+
+export type TextWrapperProp = {
+  hasImage: boolean
+};
+
+// Ability to add custom prop types in TS and styled components
+const BaseTextWrapper = styled.div.attrs<TextWrapperProp>(props => {
+  if (props.hasImage) {
+    return {
+      className: grid({ s: 9, m: 9, l: 9, xl: 9 }),
+    };
+  }
+  return {
+    className: grid({ s: 12, m: 12, l: 12, xl: 12 }),
+  };
+  
+})<TextWrapperProp>``;
+
+const MediaObjectBase = ({
+  url,
+  title,
+  labels,
+  description,
+  urlOverride,
+  extraClasses,
+  partNumber,
+  color,
+  Image,
+  DateInfo,
+  StatusIndicator,
+  ExtraInfo,
+  xOfY,
+  OverrideImageWrapper,
+  OverrideTextWrapper,
+  OverrideTitleWrapper,
+  onClick,
+} : Props) => {
+  const { x, y } = xOfY;
+  const ImageWrapper:any = OverrideImageWrapper ? OverrideImageWrapper : BaseImageWrapper;
+  const TextWrapper:any = OverrideTextWrapper ? OverrideTextWrapper : BaseTextWrapper;
+  const TitleWrapper:any = OverrideTitleWrapper ? OverrideTitleWrapper : BaseTitleWrapper;
+  const descriptionIsString = typeof description === 'string';
+  return (
+    <Space
+      v={{
+        size: 'l',
+        properties: [
+          'padding-top',
+          x === y ? undefined : 'padding-bottom',
+        ].filter(Boolean),
+      }}
+      as={url ? 'a' : 'div'}
+      href={urlOverride || url}
+      className={conditionalClassNames({
```

Comment:
confusingly – for historical reasons – there are two conditional classname functions exported from the utils file, but the one we use (almost, if not entirely exclusively) is called `classNames`, rather than `conditionalClassNames`

Reply:
I now realise that this was copy/paste. I think (from a quick find-in-repo) that this might be the _only_ place we're using this function instead of the `classNames` version.

Reply:
nice, so i can remove this function and leverage on classNames..

Reply:
found out that we need it for classNames. ClassNames uses it to call conditionalClassNames